packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -10,9 +10,342 @@
+  disabledKeys: Iterable<Key> | undefined,
+  isDisabled: boolean | undefined,
+  selectedTab: HTMLElement | undefined,
+  orientation?: Orientation,
+  density?: 'compact' | 'regular'
+}
+
+const selectedIndicator = style({
+  position: 'absolute',
+  backgroundColor: {
+    default: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'Highlight',
+      isDisabled: 'GrayText'
+    }
+  },
+  height: {
+    orientation: {
+      horizontal: '[2px]'
+    }
+  },
+  width: {
+    orientation: {
+      vertical: '[2px]'
+    }
+  },
+  bottom: {
+    orientation: {
+      horizontal: 0
+    }
+  },
+  borderStyle: 'none',
+  borderRadius: 'full'
+});
+
+const selectionAnimation = style({
+  transitionDuration: 130,
+  transitionTimingFunction: 'in-out',
+});
+
+function TabLine(props: TabLineProps) {
+  let {
+    disabledKeys,
+    isDisabled: isTabsDisabled,
+    selectedTab,
+    orientation,
+    density
+  } = props;
+  let {direction} = useLocale();
+  let state = useContext(TabListStateContext);
+
+  // We want to add disabled styling to the selection indicator only if all the Tabs are disabled
+  let [isDisabled, setIsDisabled] = useState<boolean>(false);
+  useEffect(() => {
+    let isDisabled = isTabsDisabled || isAllTabsDisabled(state?.collection || null, disabledKeys ? new Set(disabledKeys) : new Set(null));
+    setIsDisabled(isDisabled);
+  }, [state?.collection, disabledKeys, isTabsDisabled, setIsDisabled]);
+
+  let [style, setStyle] = useState<{transform: string | undefined, width: string | undefined, height: string | undefined}>({
+    transform: undefined,
+    width: undefined,
+    height: undefined
+  });
+
+  let onResize = useCallback(() => {
+    if (selectedTab) {
+      let styleObj: { transform: string | undefined, width: string | undefined, height: string | undefined } = {
+        transform: undefined,
+        width: undefined,
+        height: undefined
+      };
+
+      // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
+      let offset = direction === 'rtl' ? -1 * ((selectedTab.offsetParent as HTMLElement)?.offsetWidth - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
+      styleObj.transform = orientation === 'vertical'
+        ? `translateY(${selectedTab.offsetTop}px)`
+        : `translateX(${offset}px)`;
+
+      if (orientation === 'horizontal') {
+        styleObj.width = `${selectedTab.offsetWidth}px`;
+      } else {
+        styleObj.height = `${selectedTab.offsetHeight}px`;
+      }
+      setStyle(styleObj);
+    }
+  }, [direction, setStyle, selectedTab, orientation]);
+
+  useLayoutEffect(() => {
+    onResize();
+  }, [onResize, state?.selectedItem?.key, direction, orientation, density]);
+
+  return (
+    <div style={{...style}} className={mergeStyles(selectedIndicator({isDisabled, orientation}), selectionAnimation)} />
```

Comment:
Could the `selectionAnimation` style be merged into the definition of `selectedIndicator`? Seems like that would avoid needing to merge these at runtime